python/servo/build_commands.py

```diff
@@ -764,6 +765,33 @@ def gstreamer_root(target, env):
         return None
 
 
+def angle_root(self, target):
+    arch = {
+
+        "aarch64": "arm64",
+        "x86_64": "x64",
+    }
+    angle_arch = arch[target.split('-')[0]]
+    print("angle_x64 Architecture = {}".format(angle_arch))
+
+    angle_default_path = path.join(os.getcwd(), "support", "hololens", "packages",
+                                   "ANGLE.WindowsStore.Servo.2.1.13", "bin", "UAP", angle_arch)
+
+    # Nuget executable command
+    nuget_exe = path.join(self.msvc_package_dir('nuget'), 'nuget.exe ')
```

Comment:
Actually, given that we add the nuget directory to the PATH, we should just be able to refer to `nuget.exe` directly without using `self.msvc_package_dir`. That means we can remove the self argument from this function as well.

Reply:
Humm… I must be misunderstanging….

If I remove the nuget_exe = path.join(self.msvc_package_dir('nuget'), 'nuget.exe ') line and just refer to 'nuget.exe'  I receive the following error message.  Seems the location of nuget.exe can't be found.

I get the following error:
CalledProcessError: Command 'n u g e t . e x e   r e s t o r e   C : \ g i t - m o z i l l a \ a n g e l o r t i z 1 0 0 7 - 2 \ s u p p o r t \ h o l o l e n s \ S e r v o A p p . s l n' returned non-zero exit status 1

  File "C:\git-mozilla\angelortiz1007-2\python\servo\build_commands.py", line 294, in build
    append_to_path_env(angle_root(target_triple),
  File "C:\git-mozilla\angelortiz1007-2\python\servo\build_commands.py", line 790, in angle_root
    check_call(nuget_command)
  File "C:\git-mozilla\angelortiz1007-2\python\servo\command_base.py", line 195, in check_call
    raise subprocess.CalledProcessError(status, ' '.join(*args))

Reply:
That happens because you are still passing a string to check_call, and not a list like I proposed in my other review comments.

Reply:
Testing some related changes locally, it looks like you will need `check_call(["nuget", ...], env=env)` to allow the nuget executable to be found in the path.

Reply:
Josh, sorry I must be missing something.

I've got:
    if not os.path.exists(angle_default_path):
        check_call(['nuget.exe', 'restore', nuget_app], env)

env=env too.

I get the following error:

TypeError: bufsize must be an integer

  File "C:\git-mozilla\angelortiz1007-2\python\servo\build_commands.py", line 294, in build
    append_to_path_env(angle_root(target_triple, env), env, "LIB")
  File "C:\git-mozilla\angelortiz1007-2\python\servo\build_commands.py", line 779, in angle_root
    check_call(['nuget.exe', 'restore', nuget_app], env)
  File "C:\git-mozilla\angelortiz1007-2\python\servo\command_base.py", line 182, in check_call
    proc = subprocess.Popen(*args, shell=sys.platform == 'win32', **kwargs)
  File "C:\Python27\lib\subprocess.py", line 343, in __init__
    raise TypeError("bufsize must be an integer")

Reply:
I suspect that you do not actually have a variable named `env` at the time you are calling check_call. I recommend passing it in to angle_root as an argument; it should be available in the caller.

Reply:
Yep, I'm passing it in.  Here is my function signature...
def angle_root(target, env):
    arch = {

Here is my call to angle_root(…):
append_to_path_env(angle_root(target_triple, env), env, "LIB")

Something is missing.... Maybe in....
windows_build_env = {
    "x86_64": {
        "GSTREAMER_1_0_ROOT_X86_64": "%HOMEDRIVE%%HOMEPATH%\\gst\\gstreamer\\1.0\\x86_64\\",
    },
    "arm64": {
        "PKG_CONFIG_ALLOW_CROSS": "1",
    },
    "all": {
        "PYTHON3": "%HOMEDRIVE%%HOMEPATH%\\python3\\python.exe",
        "LINKER": "lld-link.exe",
    },
}

Reply:
So in general, the "bufsize must be an integer" error occurs when the parameters to check_call are not being passed correctly (eg. https://mihirknows.blogspot.com/2011/02/solution-raise-typeerrorbufsize-must-be.html, https://stackoverflow.com/questions/34156193/subprocess-cp-returns-error-bufsize-must-be-integer/34156248, https://stackoverflow.com/questions/17824096/bufsize-must-be-an-integer-error-while-grepping-a-message).

Reply:
Done.